### DISCUSSION TITLE
Welcome to CoLT5-attention Discussions!

### DISCUSSION THREAD

**Discussion by lucidrains** [lucidrains/CoLT5-attention#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JoaoLages** [comment]
Thanks for your open-source contribution! ❤️ 
Question: can this also be applied to a decoder-only language model like GPT?

**Reply by lucidrains** [reply]
@JoaoLages oh hey! yeah i'm trying to figure that out myself.. i may have a few ideas, but it will take a lot of time and experiments to validate

**Reply by lucidrains** [reply]
may have a solution and will build it out early next week

do you have some eval readied to see if it is any good?

**Reply by JoaoLages** [reply]
Not yet, but tell me how I can help! I'll definitely want to try to plug this type of attention in a decoder-only model! 🙌

**Reply by lucidrains** [reply]
@JoaoLages gave it a first pass https://github.com/lucidrains/CoLT5-attention/blob/main/colt5_attention/transformer_block.py#L646 will test it out probably on sunday

**Reply by lucidrains** [reply]
happy to share that experiments are a success this morning. will put more work into a triton optimized version soon.

**Reply by lucidrains** [reply]
Triton optimized version should work now! give it a try and let me know

**Reply by JoaoLages** [reply]
Great work 💪 
I hope to have time to try this by the end of this week 🤞

**Comment by n1o** [comment]
Hey @lucidrains First of I really like your work. I just have a question I recently finished reading CoLT5 paper, and I fail to see any mention of using Coordinate Descent for the Routing. Is there an reason why you used it or?

PS: I also saw you mentioned CoDA paper which says it can be used in conjunction with ColT5, but not that COO is actually used there.

Thanks for the answer :)

**Reply by n1o** [reply]
Ah I see apendix C Routing Normalization Hyperparameters. Never mind sometimes I am just too blind.
